packages/ui-color-picker/src/ColorIndicator/props.ts

```diff
@@ -0,0 +1,67 @@
+/*
+ * The MIT License (MIT)
+ *
+ * Copyright (c) 2015 - present Instructure, Inc.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+
+import PropTypes from 'prop-types'
+
+import type { WithStyleProps, ComponentStyle } from '@instructure/emotion'
+import type {
+  OtherHTMLAttributes,
+  PropValidators
+} from '@instructure/shared-types'
+
+type ColorIndicatorOwnProps = {
+  /**
+   * Valid CSS color string. E.g.: #555, rgba(55,55,55,1). It can accept empty strings
+   */
+  color?: string
+  /**
+   * Provides a reference to the `ColorIndicator`'s underlying html element.
+   */
+  elementRef?: (element: Element | null) => void
+  /**
+   * Sets the shape of the indicator. Either a circle or a rectangle
+   */
+  shape?: 'circle' | 'rectangle'
+}
+
+type PropKeys = keyof ColorIndicatorOwnProps
+
+type AllowedPropKeys = Readonly<Array<PropKeys>>
+
+type ColorIndicatorProps = ColorIndicatorOwnProps &
+  WithStyleProps<null, ColorIndicatorStyle> &
+  OtherHTMLAttributes<ColorIndicatorOwnProps>
+
+type ColorIndicatorStyle = ComponentStyle<'colorIndicator'>
+
+const propTypes: PropValidators<PropKeys> = {
+  color: PropTypes.string,
+  elementRef: PropTypes.func,
+  shape: PropTypes.oneOf(['circle', 'rectangle'])
+}
+
+const allowedProps: AllowedPropKeys = ['color']
```

Comment:
I know it is a pain to keep this up-to-date (couldn't find a good typing solution for it yet), but can you please check for all components that the `allowedProps` match the props after all the refacts?